src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -691,9 +691,12 @@ static PathElement dereferenceElement() {
      *
      * @param bitSize the padding size in bits.
      * @return the new selector layout.
-     * @throws IllegalArgumentException if {@code bitSize < 0} or {@code bitSize % 8 != 0}
+     * @throws IllegalArgumentException if {@code bitSize <= 0} or {@code bitSize % 8 != 0}
      */
     static PaddingLayout paddingLayout(long bitSize) {
+        if (bitSize <= 0) {
```

Comment:
While this does the right thing, I wonder if we can do better, as it seems we're checking size twice - perhaps we need to customize the `requireBitSizeValid` function?

Reply:
I tried that but it turned out it can be zero in other contexts. Maybe we should add back the good 'ole "allowZero" flag?

Reply:
yeah , a boolean seems good enough